spec/lib/reference.spec.js

```diff
@@ -0,0 +1,76 @@
+"use strict";
+
+describe("lib/reference", () => {
+    var ArtifactMock, ArtifactSearchMock, authToken, dateService, hostMock, instance, logger, path, Reference, refName, requestPromiseMock, URI;
+
+    ArtifactMock = require("../mock/artifact-mock")();
+    ArtifactSearchMock = require("../mock/artifact-search-mock")();
+    authToken = "abcd1234";
+    dateService = require("../../lib/date-service")();
+    hostMock = "exampleHost";
+    logger = require("../../lib/logger")();
+    path = "some/example/path";
+    refName = "exampleRefName";
+    requestPromiseMock = require("../mock/request-promise-mock")();
+    URI = require("urijs");
+    Reference = require("../../lib/reference")(ArtifactMock, ArtifactSearchMock, dateService, logger, hostMock, requestPromiseMock, URI);
+
+    beforeEach(() => {
+        instance = new Reference(refName, authToken);
+    });
+    describe(".initArtifact()", () => {
+        beforeEach(() => {
+            spyOn(instance, "buildUrl").andReturn("an/example/path");
```

Comment:
Why can't we call through to this instead? Does it make it any easier to mock? Should we be mocking functionality within a class that we are trying to test? This seems like a code smell to me. If it really requires mocking I think it should be its own module.